### ISSUE TITLE
Flow treats `async` as a reserved word—is it?

### ISSUE THREAD

**Issue by Gozala** [facebook/flow#1233]
Here is an example:

## `scratch.js`
```js
/* @flow */

import {succeed, async} from "./src/task"

```

## `stdout`

```
scratch.js:3
  3: import {succeed, async} from "./src/task"
                      ^^^^^ Unexpected token async
```

**Comment by AgentME** [comment]
ES2015 has `await` as a reserved word that can't be used as an identifier. http://www.ecma-international.org/ecma-262/6.0/#sec-reserved-words

**Comment by samwgoldman** [comment]
Woah, is it really possible that the Async Functions proposal doesn't carve out `async` as a new reserved word?

@jeffmo I don't see anything in [the proposed spec](http://tc39.github.io/ecmascript-asyncawait/). Missing something?

**Comment by mroch** [comment]
fixed by #1858
